mod/account/assets/script/toggle_visiblity.js.coffee

```diff
@@ -0,0 +1,11 @@
+$ ->
+  passwordField = $("input._toggle-password")
+  togglePassword = $("span._toggle-password")
+
+  togglePassword.on "click", () ->
```

Comment:
This is the issue, not the "open" view. The problem is when the view is loaded via ajax. I guarantee it would happen with any view.

So if you use this code, then when the document is ready (`$ ->`) it will look for all `toggle-password` spans and attach click events to them. That's great if the the span is there when the document loads. It's not great if the span doesn't appear until later via an ajax call. In that case, nothing happens to attach a click event to those spans.

We encountered this before with the attribution. The solution we used there was something like `$("body").on "click", togglePassword`. In this case the click event is attached to the body tag. It's something like "when you click on the body, check to see if you're clicking inside a password field.".  That approach works even when you use ajax.